File: juego/player.go
package juego

type Player struct {
	nombre     string
	deck       *Mazo
	cementerio *Mazo
	vida       int
	mano       []*Carta
}

func NewPlayer(name string, mazo *Mazo) *Player {
	c := NewMazo()
	return &Player{nombre: name, deck: mazo, cementerio: c, vida: 20}
}

func (p *Player) Robar() bool {
	carta := p.deck.Robar()
	if carta == nil {
		return true
	}
	p.mano = append(p.mano, carta)
	return false
}

func (p *Player) IniciarTurno() {

}

func (p *Player) GetMano() []*Carta {
	return p.mano
}

func (p *Player) GetDeck() *Mazo {
	return p.deck
}

File: juego/player.py
from juego import mazo

class Player():
    def __init__(self, name, deck):
        self.name = name
        self.deck = deck
        self.cementerio = mazo.Mazo()
        self.vida = 20
        self.mano = []

    def robar(self):
        carta = self.deck.robar()
        if carta == None:
            return True
        else:
            self.mano.append(carta)
            return False
        
    def iniciarturno(self):
        pass
    
    def getmano(self):
        # Mostrar mano del jugador
        return '\n'.join(str(carta) for carta in self.mano) if self.mano else "No tiene cartas en la mano."
